akd/src/append_only_zks.rs

```diff
@@ -196,35 +196,10 @@ impl Azks {
         storage: &StorageManager<S>,
         insertion_labels: &[NodeLabel],
     ) -> Result<u64, AkdError> {
-        let mut load_count: u64 = 0;
-        let mut current_nodes = vec![NodeKey(NodeLabel::root())];
-
-        while !current_nodes.is_empty() {
-            let nodes =
-                TreeNode::batch_get_from_storage(storage, &current_nodes, self.get_latest_epoch())
-                    .await?;
-            load_count += nodes.len() as u64;
-
-            current_nodes = Vec::<NodeKey>::new();
-
-            // Now that states are loaded in the cache, you can read and access them.
-            // Note, the two for loops are needed because otherwise, you'd be accessing remote storage
-            // individually for each node's state.
-            for node in &nodes {
-                if !Self::insertion_set_contains(insertion_labels, &node.label) {
-                    // Only continue to traverse nodes which are relevant prefixes to insertion_set
-                    continue;
-                }
-
-                for dir in crate::DIRECTIONS {
-                    let child_label = node.get_child_label(dir)?;
-                    if let Some(child_label) = child_label {
-                        current_nodes.push(NodeKey(child_label));
-                    }
-                }
-            }
-        }
-        Ok(load_count)
+        self.preload_filtered_nodes(storage, |node| {
+            !Self::insertion_set_contains(insertion_labels, &node.label)
```

Comment:
This check is backwards. Given that this isn't used to "continue" any longer, and instead filters, then this shouldn't be a NOT.